service/worker/scanner/tasklist/scavenger_test.go

```diff
@@ -193,8 +193,16 @@ func (s *ScavengerTestSuite) TestAllExpiredTasksWithErrors() {
 func (s *ScavengerTestSuite) runScavenger() {
 	s.scvgr.Start()
 	timer := time.NewTimer(scavengerTestTimeout)
+
+	waiter := make(chan struct{})
+
+	go func() {
+		s.scvgr.stopWG.Wait()
```

Comment:
I missed this during review. `s.scvgr.Stop()` is never called in tests and instead the passed ctx is expected to cancel after hardcoded 10s. Seems like that suite could be improved to manage lifecycle better and be self contained per test case.

Reply:
Kind of a fix https://github.com/uber/cadence/pull/5510